### PR TITLE
:sparkles: kirakira effect normalization

### PR DESCRIPTION
**Pull request by ChiaoteNi**
## Descriptions:
- Implement kirakira effect normalization based on the changes in the branch [kirakira_normalization] (https://github.com/cardinalblue/CBVisualEffectBuiltins/compare/dev...task/kirakira_normalization) in CBVisualEffectBuiltIn
- Also, based on the changes in the branch [kirakira_normalization](https://github.com/cardinalblue/CBVisualEffectBuiltins/compare/dev...task/kirakira_normalization), I modified some default values for the parameters as well

# Screenshots

<table>
<tr>
<th>Before</th>
<th>After</th>
</tr>
<tr>
<td>

IMG_9700

</td>
<td>

IMG_9697

</td>
</tr>
</table>

### CODE REVIEW

File: framework/Source/Kirakira.swift [hunk 1]
```diff
@@ -325,6 +340,7 @@ extension Kirakira.Parameters: Decodable {
         frameRate = try container
             .nestedContainer(keyedBy: NestedKey.self, forKey: .frameRate)
             .decode(Float.self, forKey: .value)
+        maxLength = 1000
         sparkleScale = 0.7
```
Comment (yyjim) [inline framework/Source/Kirakira.swift]: I think we shouldn't use fix values here (maxLength, sparkleScale).

File: framework/Source/Operations/CBRescaleEffect.swift [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    private func calculateTargetSize(from texture: Texture) -> CGSize {
+        let textureSize = CGSize(
+            width: CGFloat(texture.texture.width),
+            height: CGFloat(texture.texture.height)
+        )
+        let textureMaxLength = max(textureSize.width, textureSize.height)
+
+        let widthResizeRatio = maxLength > 0
+        ? CGFloat(maxLength) / textureMaxLength
+        : min(1.0, CGFloat(targetWidth) / CGFloat(textureSize.width))
```
Comment (yyjim) [inline framework/Source/Operations/CBRescaleEffect.swift:64]: nit opinion:

```suggestion
        let widthResizeRatio = maxLength > 0
            ? CGFloat(maxLength) / textureMaxLength
            : min(1.0, CGFloat(targetWidth) / CGFloat(textureSize.width))
```

Comment (yyjim) [reply framework/Source/Operations/CBRescaleEffect.swift:64]: I think we should update the documentation to set maxLength to 1 ~ 4000. The case of <0 seems like an error.

File: framework/Source/Operations/CBRescaleEffect.swift [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+        let heightResizeRatio = maxLength > 0
+        ? CGFloat(maxLength) / textureMaxLength
+        : min(1.0, CGFloat(targetHeight) / CGFloat(textureSize.height))
+
+        let width = textureSize.width * widthResizeRatio
+        let height = textureSize.height * heightResizeRatio
+
+        return CGSize(width: width, height: height)
```
Comment (yyjim) [inline framework/Source/Operations/CBRescaleEffect.swift:73]: I'm not sure if this logic is correct or not. It doesn't seem to consider keeping the texture aspect ratio.

cc @Vivienne0206

Comment (Vivienne0206) [reply framework/Source/Operations/CBRescaleEffect.swift:73]: I intended to use the assigned _targetWidth_ and _targetHeight_ as the new aspect ratio of the texture. 
Or we can just keep the _maxLength_.

File: framework/Source/Kirakira.swift [hunk 4]
```diff
@@ -196,12 +208,14 @@ public class Kirakira: OperationGroup {
...
         self.configureGroup { input, output in
             input
+            --> blendImageRescaleEffect
             --> sparklesEffect
             --> blurEffect
             --> saturationEffect
             saturationEffect.addTarget(addBlend, atTargetIndex: 1)
 
             input
+            --> sourceImageRescaleEffect
```
Comment (Vivienne0206) [inline framework/Source/Kirakira.swift]: Only apply RescaleEffect for creating sparkle images to sync the results of different image scales.
There is no need to change the input size.

Comment (ChiaoteNi) [reply framework/Source/Kirakira.swift]: Thank you for pointing this out. I'll remove it later!

File: framework/Source/Operations/CBRescaleEffect.swift (hunk 2 above)
Comment (ChiaoteNi) [reply framework/Source/Operations/CBRescaleEffect.swift:64]: `The case of <0 seems like an error `
You're right; it should be more than 1; thanks 👍

For the part of the first comment (`nit opinion`)
I just checked to format the code with `ctrl+i,` but it seems has already been Xcode's default format.

File: framework/Source/Operations/CBRescaleEffect.swift (hunk 3 above)
Comment (yyjim) [reply framework/Source/Operations/CBRescaleEffect.swift:73]: @ChiaoteNi 
I discussed this with @Vivienne0206, and she might make some adjustments to it.

Comment (ChiaoteNi) [reply framework/Source/Operations/CBRescaleEffect.swift:73]: Got it, and I'll make the same adjustments when she has done them 👍

**Pull request closed without merging**